src/hmac/index.ts

```diff
@@ -1,7 +1,12 @@
 import crypto from 'crypto'
 import lengths from './lengths.js'
 
-export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array) { // eslint-disable-line require-await
+export interface Hmac {
+  digest: (data: Uint8Array) => Promise<Uint8Array>
+  length: number
+}
+
+export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array): Promise<Hmac> {
```

Comment:
HMAC is an acronym, we capitalise acronyms elsewhere in the codebase so rename for consistency:

```suggestion
export interface HMAC {
  digest: (data: Uint8Array) => Promise<Uint8Array>
  length: number
}

export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array): Promise<HMAC> {
```